lms/product/canvas/_plugin/misc.py

```diff
@@ -3,16 +3,11 @@
 from typing import Optional
 from urllib.parse import unquote
 
-from lms.services.assignment import AssignmentService
+from lms.product.plugin.misc import MiscPlugin
 from lms.services.vitalsource import VSBookLocation
 
 
-class DocumentURLService:
-    """A service for getting LTI launch document URLs."""
-
-    def __init__(self, assignment_service: AssignmentService):
-        self._assignment_service = assignment_service
-
+class CanvasMiscPlugin(MiscPlugin):
```

Comment:
That line has not changed so I can't comment there but `_from_legacy_vitalsource_book` is Canvas only too, VS only worked in Canvas for a while·

We can eventually remove that method.

https://hypothes-is.slack.com/archives/C4K6M7P5E/p1687421709533339

Reply:
I would never, in a million years, have guessed that this was Canvas specific. I think this is a perfect example of why mixing product specific things in with normal code is so bad. It's trivial to mix in, and very difficult to spot and tease apart,